### PR TITLE
[SPARK-14465][BUILD] Checkstyle should check all Java files

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
## What changes were proposed in this pull request?

Currently, `checkstyle` is configured to check the files under `src/main/java`. However, Spark has Java files in `src/main/scala`, too. This PR fixes the following configuration in `pom.xml` and the unchecked-so-far violations on those files.

``` xml
-<sourceDirectory>${basedir}/src/main/java</sourceDirectory>
+<sourceDirectories>${basedir}/src/main/java,${basedir}/src/main/scala</sourceDirectories>
```
## How was this patch tested?

After passing the Jenkins build and manually `dev/lint-java`. (Note that Jenkins does not run `lint-java`)

### CODE REVIEW

**Comment by srowen** [comment]
Huh, why do we have .java files under the scala root? I can't think of a good reason off the top of my head.

**Comment by dongjoon-hyun** [comment]
Yes. It's strange for me, too. We may change the paths, but there is one understandable example: `package-info.java`. (It's not related to this PR, but I found 10 `package-info.java`, too.)

**Comment by dongjoon-hyun** [comment]
The following is the failure message. It seems o be irrelevant. I'll rebase this to trigger again.
```
[error] Test org.apache.spark.launcher.LauncherServerSuite.testCommunication failed: java.lang.AssertionError: null, took 0.008 sec
[error]     at org.apache.spark.launcher.LauncherServerSuite.testCommunication(LauncherServerSuite.java:89)
[error]     ...
[info] Test run finished: 1 failed, 0 ignored, 3 total, 0.141s
```

File: core/src/main/scala/org/apache/spark/io/LZ4BlockInputStream.java [hunk 1]
```diff
@@ -90,12 +87,13 @@ public LZ4BlockInputStream(InputStream in, LZ4FastDecompressor decompressor, Che
   }
 
   /**
-   * Create a new instance using {@link XXHash32} for checksuming.
+   * Create a new instance using {@link net.jpountz.xxhash.XXHash32} for checksuming.
    * @see #LZ4BlockInputStream(InputStream, LZ4FastDecompressor, Checksum)
-   * @see StreamingXXHash32#asChecksum()
+   * @see net.jpountz.xxhash.StreamingXXHash32#asChecksum()
    */
   public LZ4BlockInputStream(InputStream in, LZ4FastDecompressor decompressor) {
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/io/LZ4BlockInputStream.java]: @davies do you know why this .java file is in the scala source root? we could fix that as part of this PR if it's just an oversight

File: sql/core/src/main/scala/org/apache/spark/sql/expressions/java/typed.java [hunk 2]
```diff
@@ -28,7 +27,7 @@
 
 /**
  * :: Experimental ::
- * Type-safe functions available for {@link Dataset} operations in Java.
```
Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/expressions/java/typed.java]: @rxin same question, do you know why this wasn't under the java root?

File: core/src/main/scala/org/apache/spark/io/LZ4BlockInputStream.java (hunk 1 above)
Comment (davies) [inline core/src/main/scala/org/apache/spark/io/LZ4BlockInputStream.java]: No reason, please fix it.

File: sql/core/src/main/scala/org/apache/spark/sql/expressions/java/typed.java (hunk 2 above)
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/expressions/java/typed.java]: we can move it.

**Comment by dongjoon-hyun** [comment]
Thank you, @srowen , @davies , @rxin . I moved those files.
- rename core/src/main/{scala => java}/org/apache/spark/io/LZ4BlockInputStream.java (100%)
- rename sql/core/src/main/{scala => java}/org/apache/spark/sql/expressions/java/typed.java (100%)

File: sql/core/src/main/scala/org/apache/spark/sql/execution/BufferedRowIterator.java [hunk 3]
```diff
@@ -60,7 +60,7 @@ public long durationMs() {
   /**
    * Initializes from array of iterators of InternalRow.
    */
-  public abstract void init(int index, Iterator<InternalRow> iters[]);
+  public abstract void init(int index, Iterator<InternalRow>[] iters);
```
Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/BufferedRowIterator.java]: And we want to move this class too, right? sorry I thought that was implicit.

Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/BufferedRowIterator.java]: Oh, I see. There was some misunderstanding between us. :)  I thought there was some reason for the others.
Thank you. I'll fix right now.

**Comment by dongjoon-hyun** [comment]
@srowen . It's moved like the following.
- rename sql/core/src/main/{scala => java}/org/apache/spark/sql/execution/BufferedRowIterator.java (100%)

Now, the only 23 `package-info.java` files are under `scala`.

**Comment by srowen** [comment]
Jenkins retest this please

**Comment by dongjoon-hyun** [comment]
By the way, Jenkins fails to find `mvn`. Is there any problem there?
```
Using `mvn` from path: /home/jenkins/workspace/SparkPullRequestBuilder/build/apache-maven-3.3.9/bin/mvn
build/mvn: line 156: /home/jenkins/workspace/SparkPullRequestBuilder/build/apache-maven-3.3.9/bin/mvn: No such file or directory
Error while getting version string from Maven:
```

**Comment by srowen** [comment]
Yes, that was just fixed by Josh. (The Apache dist archives are offline)

**Comment by dongjoon-hyun** [comment]
@srowen .
Could you merge this PR?

**Comment by rxin** [comment]
Thanks - merging in master.

**Comment by dongjoon-hyun** [comment]
Thank you for merging, @rxin . 
Jenkins seems to have some trouble during merging. :)

**Pull request closed without merging**